### PR TITLE
Tpi 987/master

### PR DESCRIPTION
**Pull request by hakiemaul**

### CODE REVIEW

File: Xendit/M2Invoice/Model/Payment/AbstractInvoice.php [hunk 1]
```diff
@@ -108,6 +108,10 @@ public function isAvailable(\Magento\Quote\Api\Data\CartInterface $quote = null)
             return true;
         }
 
+        // if ($this->methodCode === 'OVO') {
```
Comment (candrasaputra) [inline Xendit/M2Invoice/Model/Payment/AbstractInvoice.php]: delete this comment

File: Xendit/M2Invoice/Model/Payment/OVO.php [hunk 2]
```diff
@@ -81,17 +102,62 @@ private function requestEwalletPayment($requestData)
...
+     * @param $ewalletType
+     * @param $externalId
+     * @return array
+     */
+    private function getEwalletPaymentStatus($ewalletType, $externalId)
+    {
+        $ewalletUrl = $this->dataHelper->getCheckoutUrl() . "/payment/xendit/ewallets?ewallet_type=" . $ewalletType . "&external_id=" . $externalId;
+        $ewalletMethod = \Zend\Http\Request::METHOD_GET;
+
+        // return array(
```
Comment (candrasaputra) [inline Xendit/M2Invoice/Model/Payment/OVO.php]: delete this comment

File: Xendit/M2Invoice/Model/Payment/OVO.php [hunk 3]
```diff
@@ -115,6 +186,13 @@ private function getPaymentMethod()
...
+    /**
+     * Retrieved specified element from given array.
+     *
+     * @param $data array
+     * @param $element string
+     * @return string
+     */
     private function elementFromArray($data, $element)
     {
         $r = [];
```
Comment (candrasaputra) [inline Xendit/M2Invoice/Model/Payment/OVO.php]: I think it more readable if name variable is $result or $temporary.

File: Xendit/M2Invoice/Model/Payment/AbstractInvoice.php (hunk 1 above)
Comment (candrasaputra) [inline Xendit/M2Invoice/Model/Payment/AbstractInvoice.php]: Delete this

File: Xendit/M2Invoice/Model/Payment/OVO.php (hunk 2 above)
Comment (candrasaputra) [inline Xendit/M2Invoice/Model/Payment/OVO.php]: delete this

File: Xendit/M2Invoice/Model/Payment/OVO.php [hunk 4]
```diff
@@ -68,6 +83,12 @@ public function authorize(\Magento\Payment\Model\InfoInterface $payment, $amount
         return $this;
     }
 
+    /**
```
Comment (candrasaputra) [inline Xendit/M2Invoice/Model/Payment/OVO.php]: cool.. we have method description now..

Comment (hakiemaul) [reply Xendit/M2Invoice/Model/Payment/OVO.php]: lol one step at a time

**Comment by hakiemaul** [comment]
closed, change approach to callback based, see #17

**Pull request closed without merging**